core/model/src/main/java/org/eclipse/rdf4j/model/util/Literals.java

```diff
@@ -74,7 +74,8 @@ public static String getLabel(Optional v, String fallback) {
 		if (l instanceof SimpleLiteral) {
 			return ((SimpleLiteral) l).getXsdDatatype();
 		}
-		return Optional.empty();
+		// we return null to signify that the literal does not support the new XSD Datatype enum
+		return null;
```

Comment:
Using null to signal lack of support for XSD datatype.

I was thinking about maybe using an interface for the XSD datatype enum, but then we won't be able to use `==`. The only real solution is to put the XSD class somewhere so we can depend on it from both the Literal class and the code that does sorting.

Reply:
Out of scope for this PR.

Reply:
https://github.com/eclipse/rdf4j/issues/3587

Reply:
I am not really following why you are letting a method that has an optional as it return type return `null`. That seems error-prone to me.

Reply:
The point is for the method to return an empty optional if the datatype isn't an XSD datatype. But it also used to return an empty optional if the support for XSD datatypes was missing. So an empty optional would signal both "this is not an XSD datatype" and "I don't know if it's an XSD datatype".

Reply:
Ah I see, thanks. That makes sense then. We should reflect this distinction in the javadoc though.

Reply:
Yeah. Definitely. Though I have the issue mentioned above where I want to add support for core datatypes as enums directly in the Literal interface and force other implementers to support it too. I that would really simplify our handling of datatypes pretty much everywhere. 

Something like `Optional<CoreDatatype> getCoreDatatype()` and have CoreDatatype within the model-api module.